backend/person/models.py

```diff
@@ -1,36 +1,64 @@
 from django.db import models
+from django.forms import ValidationError
 from core.models import Field, LettercraftDate
+from django.db.models import Q, CheckConstraint
 
 
-class Office(models.Model):
+class StatusMarker(models.Model):
     """
-    A job or position that a person can hold.
+    A marker of someone's social status. This can be a job title or a social group someone belongs to.
     """
 
     name = models.CharField(
         max_length=256,
-        help_text="The name of the office (e.g. 'bishop of Toulouse', 'king of France')",
+        help_text="The name of the marker or job (e.g. 'bishop of Toulouse', 'peasantry of France')",
     )
     description = models.TextField(
         blank=True,
-        help_text="A description of the office (e.g. 'The bishop of Toulouse is the ecclesiastical ruler of the diocese of Toulouse')",
+        help_text="A description of what the marker means (e.g. 'The bishop of Toulouse is the ecclesiastical ruler of the diocese of Toulouse')",
     )
 
     def __str__(self):
         return self.name
 
 
-class Person(models.Model):
-    class Gender(models.TextChoices):
-        FEMALE = "FEMALE", "Female"
-        MALE = "MALE", "Male"
-        UNKNOWN = "UNKNOWN", "Unknown"
-        OTHER = "OTHER", "Other"
+class Gender(models.TextChoices):
+    FEMALE = "FEMALE", "Female"
+    MALE = "MALE", "Male"
+    UNKNOWN = "UNKNOWN", "Unknown"
+    MIXED = "MIXED", "Mixed"
+    OTHER = "OTHER", "Other"
+
 
+class Person(models.Model):
     gender = models.CharField(
-        max_length=8, choices=Gender.choices, default=Gender.UNKNOWN
+        max_length=8,
+        choices=Gender.choices,
+        default=Gender.UNKNOWN,
+        help_text="The gender of this person or the group members. The option Mixed is only used for groups.",
     )
 
+    is_group = models.BooleanField(
+        default=False,
+        help_text="Check if this entity is a group of people (e.g. 'the nuns of Poitiers'). If checked, leave the Date of Birth and Date of Death fields empty.",
+    )
+
+    class Meta:
+        constraints = [
+            CheckConstraint(
+                check=~Q(gender=Gender.MIXED, is_group=True),
+                name="gender_group_constraint",
+                violation_error_message="The 'mixed' gender option is reserved for groups",
+            )
+        ]
+
+    def clean(self):
```

Comment:
The 'mixed-gender-cannot-go-on-groups'-constraint defined in the Meta is a database constraint. I tried to do the same for the `no-date-of-birth/death-for-a-group' constraints, but since they involve foreign relations, the database cannot enforce them. That is why these had to be put in `clean()` instead.